.github/actions/vtk-install-dep/action.yml

```diff
@@ -0,0 +1,116 @@
+name: 'Install VTK Dependency'
+description: 'Install VTK Dependency using cache when possible'
+inputs:
+  vtk_version:
+    description: 'VTK version'
+    required: false
+    default: 'commit'
+  vtk_sha_file:
+    description: 'VTK SHA file'
+    required: false
+    default: ''
+  os:
+    description: 'OS to build on'
+    required: false
+    default: 'ubuntu-latest'
+runs:
+  using: "composite"
+  steps:
+
+    - name: Recover VTK Short SHA from file
+      if: |
+        steps.cache-vtk.outputs.cache-hit != 'true' &&
+        inputs.vtk_version == 'commit'
+      working-directory: ${{github.workspace}}
+      shell: bash
+      run: echo "VTK_SHA_OR_TAG=$(<${{ inputs.vtk_sha_file }})" >> $GITHUB_ENV
+
+    - name: Copy VTK version to env var
+      if: |
+        steps.cache-vtk.outputs.cache-hit != 'true' &&
+        inputs.vtk_version != 'commit'
+      working-directory: ${{github.workspace}}
+      shell: bash
+      run: echo "VTK_SHA_OR_TAG=${{inputs.vtk_version}}" >> $GITHUB_ENV
```

Comment:
this step looks very similar to the previous one

Reply:
Yes it is different. Check the if condition on top

Reply:
I see